### PR TITLE
blob images to base64 data on html export

### PR DESCRIPTION
**Pull request by shashankp**
https://github.com/fonsp/Pluto.jl/issues/380

### CODE REVIEW

**Comment by fonsp** [comment]
Nice, thanks!

The function is async now, does that mean that if you are fast enough, the download link does not work?

**Comment by fonsp** [comment]
Or maybe we can just assume that no one is that fast

**Review by fonsp — CHANGES_REQUESTED** [review]
https://github.com/fonsp/Pluto.jl/pull/384#issuecomment-688728734

File: frontend/components/Editor.js [hunk 1]
```diff
@@ -765,16 +765,23 @@ export class Editor extends Component {
...
+                                    download_link.href = URL.createObjectURL(
+                                        new Blob([html], {
                                             type: "text/html",
                                         })
                                     )
+                                    document.body.appendChild(download_link);               
+                                    download_link.click();                                   
+                                    document.body.removeChild(download_link); 
+                                }).finally(() => {
+                                    export_card.classList.remove("export_in_progress");
```
Comment (shashankp) [inline frontend/components/Editor.js:783]: For large files / online notebooks, user can be given visual feedback

Comment (fonsp) [reply frontend/components/Editor.js:783]: Thanks for this addition! 

Even if it is online, the blobs are already in memory (right?), so the overhead should be tiny.

I will get a big notebook to test this one

File: frontend/common/OfflineHTMLExport.js [hunk 2]
```diff
@@ -1,27 +1,60 @@
...
-    <html lang="en">
-    <head>
-        <meta name="viewport" content="width=device-width" />
-        <title>⚡ Pluto.jl ⚡</title>
-        <meta charset="utf-8" />
+    var blob_to_base64_promises = [];
+    Array.from(body.querySelectorAll("img")).forEach((img) => {
+        if (img.src.match(/^blob:/)) {            
+            blob_to_base64_promises.push(new Promise((resolve) => {
+                var xhr = new XMLHttpRequest();
```
Comment (fonsp) [inline frontend/common/OfflineHTMLExport.js:11]: I found this SO answer, looks like we can use `fetch` instead - that would be easier on the eyes

https://stackoverflow.com/questions/11876175/how-to-get-a-file-or-blob-from-an-object-url

I'll try this myself

**Comment by fonsp** [comment]
Sorry I clicked on "Request changes" but I will look it this myself

**Comment by shashankp** [comment]
> Sorry I clicked on "Request changes" but I will look it this myself

Please check the added commits to get around the raised issue.

**Comment by fonsp** [comment]
Thanks, it's merged! I'm a git noob so i made a new PR with your commits: https://github.com/fonsp/Pluto.jl/pull/457

**Pull request closed without merging**